application/shoot.c

```diff
@@ -459,39 +451,53 @@ static void trigger_motor_turn_back(void)
 	}
 }
 
-
-static void shoot_bullet_control(void)
+static void load_bullet_control(void)
 {
     // Rotate by TRIGGER_ANGLE_INCREMENT every time
-    if (shoot_control.move_flag == 0)
-    {
-        shoot_control.set_angle = rad_format(shoot_control.angle + TRIGGER_ANGLE_INCREMENT);
-        shoot_control.move_flag = 1;
-    }
+    
 
-    if ((shoot_control.press_l == 0) && (shoot_control.shoot_rc->rc.s[RC_LEFT_LEVER_CHANNEL] != RC_SW_UP) && (shoot_control.key == SWITCH_TRIGGER_OFF))
+    if ((shoot_control.press_r == 0) && (shoot_control.shoot_rc->rc.s[RC_LEFT_LEVER_CHANNEL] != RC_SW_UP) && (shoot_control.key == SWITCH_TRIGGER_OFF))
     {
         shoot_control.shoot_mode = SHOOT_STOP;
     }
+    if((shoot_control.press_r == 1)&&(shoot_control.press_l == 0))
+    {
+        shoot_control.trigger_speed_set = 0;
+        shoot_control.shoot_mode = SHOOT_READY_FRIC;
+    }
 
 #if SHOOT_WITH_REF_DATA
     get_shoot_heat_limit_and_heat(&shoot_control.heat_limit, &shoot_control.heat);
-    if (!toe_is_error(REFEREE_TOE) && (shoot_control.heat + SHOOT_HEAT_REMAIN_VALUE > shoot_control.heat_limit))
+    if (!toe_is_error(REFEREE_TOE) && (shoot_control.heat + SHOOT_HEAT_REMAIN_VALUE > shoot_control.heat_limit ))
     {
         shoot_control.trigger_speed_set = 0;
     }
     else
 #endif
-	{
-		if (rad_format(shoot_control.set_angle - shoot_control.angle) > 0.05f)
-		{
-			shoot_control.trigger_speed_set = SEMI_AUTO_FIRE_TRIGGER_SPEED;
-		}
-		else
-		{
-			shoot_control.trigger_speed_set = 0.0f;
-			shoot_control.move_flag = 0;
-		}
-	}
+    {
+    if(shoot_control.shoot_mode == SHOOT_SEMI_AUTO_FIRE)
+    {    
+        if (shoot_control.move_flag == 0)
+        {
+            shoot_control.set_angle = rad_format(shoot_control.angle + TRIGGER_ANGLE_INCREMENT);
+            shoot_control.move_flag = 1;
+        }
+        if (rad_format(shoot_control.set_angle - shoot_control.angle) > 0.05f)
+        {
+            shoot_control.trigger_speed_set = SEMI_AUTO_FIRE_TRIGGER_SPEED;
+        }
+        else
+        {
+            shoot_control.trigger_speed_set = 0.0f;
+            shoot_control.move_flag = 0;
+            shoot_control.shoot_mode = SHOOT_READY_FRIC;
+        }    
+    }
+    else if(shoot_control.shoot_mode == SHOOT_AUTO_FIRE)
+    {
+        shoot_control.trigger_speed_set = AUTO_FIRE_TRIGGER_SPEED;
+    }
```

Comment:
The old way to set `trigger_speed_set` was better, in that it would only do the assignment once at the entering edge of state.